indico/modules/auth/providers.py

```diff
@@ -5,20 +5,21 @@
 # modify it under the terms of the MIT License; see the
 # LICENSE file for more details.
 
-from flask import session
-from flask_multipass.providers.sqlalchemy import SQLAlchemyAuthProviderBase, SQLAlchemyIdentityProviderBase
+from flask import flash, session
+from flask_multipass import AuthInfo, AuthProvider, InvalidCredentials, NoSuchUser
+from flask_multipass.providers.sqlalchemy import SQLAlchemyIdentityProviderBase
 
+from indico.core.db import db
 from indico.modules.auth import Identity, logger
 from indico.modules.auth.forms import LocalLoginForm
 from indico.modules.users import User
+from indico.modules.users.models.emails import UserEmail
+from indico.util.i18n import _
 from indico.util.passwords import validate_secure_password
 
 
-class IndicoAuthProvider(SQLAlchemyAuthProviderBase):
+class IndicoAuthProvider(AuthProvider):
```

Comment:
I think we also need a unit test for the new logic considering this is rather sensitive logic.

This could either be done by using the test client and simulating actual requests, or simply by instantiating the `IndicoAuthProvider`, mocking out its `multipass` attribute, and then calling its `process_local_login` method and asserting that the mocked `self.multipass.handle_auth_success(auth_info)` has been called (in case of a successful login only)

Reply:
Okay, will add tests for this

Reply:
Done